cli/handleSysCheck.js

```diff
@@ -13,13 +23,115 @@ const renderHelp = () => `
     $ yeep syscheck --config=yeep.config.js
 `;
 
+const validateMongo = async (uri, spinner) => {
+  const client = await MongoClient.connect(
+    uri,
+    { useNewUrlParser: true }
+  );
+
+  spinner.stopAndPersist({
+    symbol: chalk.green('✔'),
+    text: 'Connected to database',
+  });
+
+  const db = client.db();
+  try {
+    // TODO: Verify that the uri has the username:password format if the authentication fails!
+    const buildInfo = await db.admin().buildInfo();
+    const isVersionValid = buildInfo.versionArray[0] >= 4;
+    if (!isVersionValid) {
+      spinner.fail(`Mongo version ${buildInfo.version} needs to be higher than 4.0.0`);
+      throw new Error('Mongo version missmatch');
+    }
+    spinner.stopAndPersist({
+      symbol: chalk.green('✔'),
+      text: `Mongo version ${buildInfo.version} is valid (>= 4.0.0)`,
+    });
+
+    let replStatus;
+    try {
+      replStatus = await db.admin().replSetGetStatus();
+    } catch (err) {
+      spinner.fail(`Mongo database needs to exist within a replica set`);
+      throw err;
+    }
+    spinner.stopAndPersist({
+      symbol: chalk.green('✔'),
+      text: `Mongo replica set ${replStatus.set} found`,
+    });
+
```

Comment:
At this point you should close the client connection, as it's not needed anymore.